### ISSUE TITLE
Weird disortions in panorama image

### ISSUE THREAD

**Issue by Skayo** [mistic100/Photo-Sphere-Viewer#631]
Hey there!

I have some weird disortions (wavy-lines) in my panorama images, that should not appear, and was wondering if you know why they happen or what I can do against this.
Here's a screenshot of my panorama using `photo-sphere-viewer`:
![mine](https://user-images.githubusercontent.com/10259118/151195210-769753fd-48ec-4d9e-8fe4-d477a48c70d1.png)

And here, in comparison, the exact same image uploaded to theasys.io:
![theasys](https://user-images.githubusercontent.com/10259118/151195274-2369ad6d-1606-4e54-a251-3c2a4bc4e4e9.png)

Do you have any ideas what causes this?
Feel free to ask if you need any more info.

### Versions:
- Photo-Sphere-Viewer: `4.4.3`
- three.js: `0.136.0`
- Chrome: `Version 97.0.4692.99 (Official Build) (arm64)`

**Comment by mistic100** [comment]
I cannot answer without an online demo, or at least your entire panorama file.

**Comment by Skayo** [comment]
Is there a way I can send it to you privately?

**Comment by mistic100** [comment]
You can find my email in my commits : contact@git.strangeplanet.fr

**Comment by mistic100** [comment]
So I took a look at your panoramas and unfortunately there is not solution. This is inheritent by the rendering technique used. Other softwares my use clever rendering technics that are way ahead my knowledge (and would require to rewrite everything without THREE.js).

**Comment by mistic100** [comment]
I fact it can be mitigated by increasing this constant https://github.com/mistic100/Photo-Sphere-Viewer/blob/dev/src/adapters/equirectangular/index.js#L9

But I am afraid of negative performances impacts.

**Comment by Skayo** [comment]
Bummer.... Well, thanks a lot for taking a look at it anyways! I'll keep investigating into the problem, otherwise I'll have to find another library or just write my own.

**Comment by mistic100** [comment]
Try it by increasing `SPHERE_SEGMENTS` (must be power of 2) in your local copy of the lib. If you are okay with the performances, it can become an option.

**Comment by mistic100** [comment]
are you sure you changed the right value ? I **did** see a difference with `128`

**Comment by Skayo** [comment]
> are you sure you changed the right value and cleared your cache ? I **did** see a difference with `128`

hmmm I'll check again

**Comment by Skayo** [comment]
Omg yeah I do see a difference! Weird...
Earlier I forgot to clear the cache used by my bundler, so I didn't actually update the `SPHERE_SEGMENTS`.
Anyway, I guess it could be pretty useful to be able to specify this value in the config or something

**Comment by mistic100** [comment]
The configuration in version 4.5.0 will be :

```js
new Viewer({
  adapter: [EquirectangularAdapter, {
    resolution: 128,
  }],
  panorama: 'path/panorama.jpg',
});
```

**Comment by Skayo** [comment]
Okay so I did a [little research](https://onix-systems.medium.com/how-to-use-360-equirectangular-panoramas-for-greater-realism-in-games-55fadb0547daand) and it turns out that using equirectangular images causes quite some data redundancy on the poles (top / bottom). Therefore I tried using cube maps instead, et voilà, no more weird distortions or anything, and it actually looks a lot better because the texture on the poles isn't squished together!

I just noticed that you have already implemented the new segment configuration feature, and I'm sure it doesn't hurt to have that feature anyway, even if I don't need it anymore. So thanks a lot!
